module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,14.27,True,2023-06-07T07:00:21,,
cowidev.vax.batch.australia,,,2023-06-07T07:00:06,,
cowidev.vax.batch.austria_old,,,2023-06-07T07:00:06,,
cowidev.vax.batch.belgium,8.25,True,2023-06-07T07:00:15,,
cowidev.vax.batch.bolivia,4.02,True,2023-06-07T07:00:10,,
cowidev.vax.batch.canada,6.04,False,2023-06-07T07:01:57,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 282, in main
    Canada().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 270, in export
    df = df.pipe(self.pipeline, df_man)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 250, in pipeline
    .pipe(self.pipe_make_monotonic)[
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 230, in pipe_make_monotonic
    df = df.pipe(self.make_monotonic, max_removed_rows=self.max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
",1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
cowidev.vax.batch.chile,0.9,True,2023-06-07T07:00:07,,
cowidev.vax.batch.czechia,23.8,True,2023-06-07T07:00:30,,
cowidev.vax.batch.denmark,,,2023-06-07T07:00:06,,
cowidev.vax.batch.ecdc,86.38,True,2023-06-07T07:01:33,,
cowidev.vax.batch.ecuador,,,2023-06-07T07:00:06,,
cowidev.vax.batch.estonia,0.82,True,2023-06-07T07:00:07,,
cowidev.vax.batch.france,6.37,True,2023-06-07T07:00:13,,
cowidev.vax.batch.germany,0.17,True,2023-06-07T07:00:06,,
cowidev.vax.batch.greece,23.88,True,2023-06-07T07:00:30,,
cowidev.vax.batch.hong_kong,2.79,False,2023-06-07T07:01:59,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 224, in main
    HongKong().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 208, in export
    df = df_base.pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 117, in pipeline
    .pipe(self.make_monotonic)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 225 would be NaNed for metric total_vaccinations. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-05-20']. Location(s) affected: {'Hong Kong'}
",1 chunks of lengths 225 would be NaNed for metric total_vaccinations. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-05-20']. Location(s) affected: {'Hong Kong'}
cowidev.vax.batch.indonesia,,,2023-06-07T07:00:06,,
cowidev.vax.batch.ireland,2.39,False,2023-06-07T07:02:02,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/ireland.py"", line 135, in main
    Ireland().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/ireland.py"", line 130, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/ireland.py"", line 27, in read
    return pd.merge(data_primary, data_booster, how=""outer"", on=""date"", validate=""one_to_one"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/reshape/merge.py"", line 106, in merge
    op = _MergeOperation(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/reshape/merge.py"", line 709, in __init__
    self._validate(validate)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/reshape/merge.py"", line 1429, in _validate
    raise MergeError(
pandas.errors.MergeError: Merge keys are not unique in right dataset; not a one-to-one merge
",Merge keys are not unique in right dataset; not a one-to-one merge
cowidev.vax.batch.israel,7.79,True,2023-06-07T07:00:14,,
cowidev.vax.batch.italy,31.51,True,2023-06-07T07:00:38,,
cowidev.vax.batch.japan,9.85,False,2023-06-07T07:02:12,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 375, in main
    Japan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 359, in export
    df = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 71, in read
    df_latest = self.read_latest().pipe(self.pipe_read_latest)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 100, in read_latest
    self._read_latest_booster(),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 211, in _read_latest_booster
    df = self._read_xlsx(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 233, in _read_xlsx
    raise ValueError(f""Unknown sheets: {sheets_unknown}"")
ValueError: Unknown sheets: {'5.8以降接種_総接種回数', '5.8以降接種_一般接種（４回目）', '６回目_一般接種回数', '５回目_一般接種回数', '5.8以降接種_一般接種（６回目）', '5.8以降接種_一般接種（３回目）', '５回目_総接種回数', '６回目_総接種回数', '5.8以降接種_一般接種（５回目）'}
","Unknown sheets: {'5.8以降接種_総接種回数', '5.8以降接種_一般接種（４回目）', '６回目_一般接種回数', '５回目_一般接種回数', '5.8以降接種_一般接種（６回目）', '5.8以降接種_一般接種（３回目）', '５回目_総接種回数', '６回目_総接種回数', '5.8以降接種_一般接種（５回目）'}"
cowidev.vax.batch.jersey,5.32,True,2023-06-07T07:00:12,,
cowidev.vax.batch.latvia,40.23,True,2023-06-07T07:00:46,,
cowidev.vax.batch.lithuania,3.96,True,2023-06-07T07:00:10,,
cowidev.vax.batch.luxembourg,3.11,True,2023-06-07T07:00:09,,
cowidev.vax.batch.malaysia,0.66,True,2023-06-07T07:00:07,,
cowidev.vax.batch.malta,1.06,True,2023-06-07T07:00:07,,
cowidev.vax.batch.netherlands,,,2023-06-07T07:00:06,,
cowidev.vax.batch.new_zealand,12.99,True,2023-06-07T07:00:19,,
cowidev.vax.batch.norway,0.52,True,2023-06-07T07:00:07,,
cowidev.vax.batch.peru,4.45,True,2023-06-07T07:00:11,,
cowidev.vax.batch.portugal,,,2023-06-07T07:00:06,,
cowidev.vax.batch.romania,12.13,True,2023-06-07T07:00:18,,
cowidev.vax.batch.saudi_arabia,9.33,True,2023-06-07T07:00:16,,
cowidev.vax.batch.singapore,4.56,True,2023-06-07T07:00:11,,
cowidev.vax.batch.slovakia,48.66,True,2023-06-07T07:00:55,,
cowidev.vax.batch.slovenia,2.05,True,2023-06-07T07:00:08,,
cowidev.vax.batch.south_korea,21.34,False,2023-06-07T07:02:33,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 209, in main
    SouthKorea().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 191, in export
    df_base = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 109, in pipeline_base
    .pipe(self.pipe_check_metrics)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 92, in pipe_check_metrics
    raise ValueError(f""Metric {dose} for 'all' is not equal to the sum over all vaccines (2% tolerance)."")
ValueError: Metric dose_1 for 'all' is not equal to the sum over all vaccines (2% tolerance).
",Metric dose_1 for 'all' is not equal to the sum over all vaccines (2% tolerance).
cowidev.vax.batch.spc,31.43,True,2023-06-07T07:00:38,,
cowidev.vax.batch.sweden,12.42,True,2023-06-07T07:00:19,,
cowidev.vax.batch.switzerland,72.69,True,2023-06-07T07:01:20,,
cowidev.vax.batch.trinidad_and_tobago,3.65,True,2023-06-07T07:00:11,,
cowidev.vax.batch.ukraine,,,2023-06-07T07:00:07,,
cowidev.vax.batch.united_kingdom,14.39,True,2023-06-07T07:00:22,,
cowidev.vax.batch.united_states,15.48,True,2023-06-07T07:00:23,,
cowidev.vax.batch.uruguay,5.64,True,2023-06-07T07:00:14,,
cowidev.vax.batch.zimbabwe,3.11,True,2023-06-07T07:00:12,,
cowidev.vax.incremental.africacdc,,,2023-06-07T07:00:10,,
cowidev.vax.incremental.antigua_barbuda,,,2023-06-07T07:00:10,,
cowidev.vax.incremental.aruba,1.42,True,2023-06-07T07:00:11,,
cowidev.vax.incremental.austria,,,2023-06-07T07:00:10,,
cowidev.vax.incremental.azerbaijan,,,2023-06-07T07:00:10,,
cowidev.vax.incremental.bahrain,,,2023-06-07T07:00:10,,
cowidev.vax.incremental.bangladesh,49.29,True,2023-06-07T07:01:00,,
cowidev.vax.incremental.barbados,,,2023-06-07T07:00:10,,
cowidev.vax.incremental.brazil,0.94,True,2023-06-07T07:00:11,,
cowidev.vax.incremental.bulgaria,5.98,True,2023-06-07T07:00:17,,
cowidev.vax.incremental.china,20.11,True,2023-06-07T07:00:31,,
cowidev.vax.incremental.costa_rica,,,2023-06-07T07:00:11,,
cowidev.vax.incremental.croatia,1.54,True,2023-06-07T07:00:12,,
cowidev.vax.incremental.cuba,9.45,True,2023-06-07T07:00:20,,
cowidev.vax.incremental.curacao,,,2023-06-07T07:00:11,,
cowidev.vax.incremental.cyprus,2.89,True,2023-06-07T07:00:14,,
cowidev.vax.incremental.dominican_republic,13.02,False,2023-06-07T07:02:46,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 112, in main
    DominicanRepublic().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 26, in read
    metrics = self._parse_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 35, in _parse_metrics
    metric_candidates = _find_potential_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 91, in _find_potential_metrics
    h3 = _find_h3(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 81, in _find_h3
    raise ValueError(""More than one element found!"")
ValueError: More than one element found!
",More than one element found!
cowidev.vax.incremental.el_salvador,8.5,True,2023-06-07T07:00:21,,
cowidev.vax.incremental.faeroe_islands,1.52,True,2023-06-07T07:00:14,,
cowidev.vax.incremental.fiji,0.0,True,2023-06-07T07:00:13,,
cowidev.vax.incremental.finland,0.4,True,2023-06-07T07:00:13,,
cowidev.vax.incremental.gabon,,,2023-06-07T07:00:13,,
cowidev.vax.incremental.georgia,5.24,True,2023-06-07T07:00:18,,
cowidev.vax.incremental.greenland,,,2023-06-07T07:00:14,,
cowidev.vax.incremental.guatemala,,,2023-06-07T07:00:14,,
cowidev.vax.incremental.guernsey,,,2023-06-07T07:00:14,,
cowidev.vax.incremental.hungary,65.77,True,2023-06-07T07:01:20,,
cowidev.vax.incremental.iceland,1.58,True,2023-06-07T07:00:16,,
cowidev.vax.incremental.india,3.42,False,2023-06-07T07:02:49,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/india.py"", line 100, in main
    India().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/india.py"", line 86, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/india.py"", line 27, in read
    data = request_json(self.source_url[self.source_name])
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 106, in request_json
    text = request_text(url, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 127, in request_text
    soup = get_soup(url, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 83, in get_soup
    response = get_response(source, request_method, use_proxy, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 52, in get_response
    raise ValueError(
ValueError: Source https://api.cowin.gov.in/api/v1/reports/v2/getPublicReports?state_id=&district_id=&date=2023-06-07 not reached! Error code 400 Bad Request
",Source https://api.cowin.gov.in/api/v1/reports/v2/getPublicReports?state_id=&district_id=&date=2023-06-07 not reached! Error code 400 Bad Request
cowidev.vax.incremental.iran,,,2023-06-07T07:00:14,,
cowidev.vax.incremental.isle_of_man,2.33,True,2023-06-07T07:00:16,,
cowidev.vax.incremental.jamaica,8.78,True,2023-06-07T07:00:23,,
cowidev.vax.incremental.kazakhstan,7.39,True,2023-06-07T07:00:23,,
cowidev.vax.incremental.kosovo,,,2023-06-07T07:00:16,,
cowidev.vax.incremental.kyrgyzstan,5.81,True,2023-06-07T07:00:21,,
cowidev.vax.incremental.laos,1.66,True,2023-06-07T07:02:51,,
cowidev.vax.incremental.macao,6.64,False,2023-06-07T07:02:58,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 90, in main
    Macao().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 75, in export
    data = self.read()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 23, in read
    iframe_url = self._get_iframe_url(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 31, in _get_iframe_url
    elem = driver.find_element_by_id(""ICovid19Monitor"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 360, in find_element_by_id
    return self.find_element(by=By.ID, value=id_)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 321, in execute
    self.error_handler.check_response(response)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py"", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)

","Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)
"
cowidev.vax.incremental.moldova,,,2023-06-07T07:00:17,,
cowidev.vax.incremental.monaco,78.46,True,2023-06-07T07:01:35,,
cowidev.vax.incremental.mongolia,1.37,True,2023-06-07T07:00:18,,
cowidev.vax.incremental.montenegro,0.96,True,2023-06-07T07:00:19,,
cowidev.vax.incremental.morocco,,,2023-06-07T07:00:18,,
cowidev.vax.incremental.myanmar,,,2023-06-07T07:00:18,,
cowidev.vax.incremental.nepal,,,2023-06-07T07:00:18,,
cowidev.vax.incremental.north_macedonia,,,2023-06-07T07:00:18,,
cowidev.vax.incremental.northern_cyprus,4.71,True,2023-06-07T07:00:23,,
cowidev.vax.incremental.paho,85.42,False,2023-06-07T07:04:23,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 195, in main
    PAHO().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 190, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 167, in pipeline
    df.pipe(self.pipe_check_columns)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 122, in pipe_check_columns
    raise ValueError(f""Missing column fields: {columns_missing}. Present columns are: {df.columns}"")
ValueError: Missing column fields: {'1st additional dose', 'Total doses', 'Second dose', 'First dose', 'Country code', 'Country/ Territory', 'Single dose', '2nd additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')
","Missing column fields: {'1st additional dose', 'Total doses', 'Second dose', 'First dose', 'Country code', 'Country/ Territory', 'Single dose', '2nd additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')"
cowidev.vax.incremental.pakistan,,,2023-06-07T07:00:18,,
cowidev.vax.incremental.philippines,,,2023-06-07T07:00:18,,
cowidev.vax.incremental.poland,2.03,True,2023-06-07T07:00:20,,
cowidev.vax.incremental.qatar,,,2023-06-07T07:00:19,,
cowidev.vax.incremental.russia,18.94,True,2023-06-07T07:00:38,,
cowidev.vax.incremental.saint_lucia,,,2023-06-07T07:00:19,,
cowidev.vax.incremental.serbia,5.0,True,2023-06-07T07:00:24,,
cowidev.vax.incremental.south_africa,,,2023-06-07T07:00:19,,
cowidev.vax.incremental.spain,24.31,True,2023-06-07T07:00:44,,
cowidev.vax.incremental.sri_lanka,,,2023-06-07T07:00:20,,
cowidev.vax.incremental.suriname,15.21,True,2023-06-07T07:00:36,,
cowidev.vax.incremental.taiwan,5.54,False,2023-06-07T07:04:29,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 184, in main
    Taiwan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 170, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 38, in read
    data = self.parse_data(df, soup)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 122, in parse_data
    ""date"": self._parse_date(soup),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 156, in _parse_date
    date_str = re.search(regex, date_raw).group(1)
AttributeError: 'NoneType' object has no attribute 'group'
",'NoneType' object has no attribute 'group'
cowidev.vax.incremental.thailand,4.96,True,2023-06-07T07:00:25,,
cowidev.vax.incremental.turkey,,,2023-06-07T07:00:21,,
cowidev.vax.incremental.united_arab_emirates,,,2023-06-07T07:00:21,,
cowidev.vax.incremental.vietnam,,,2023-06-07T07:00:21,,
cowidev.vax.incremental.who,4.96,True,2023-06-07T07:00:26,,
cowidev.vax.incremental.zambia,1.13,True,2023-06-07T07:00:23,,
